module/VuFind/src/VuFind/Db/Service/UserListService.php

```diff
@@ -0,0 +1,414 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+
+namespace VuFind\Db\Service;
+
+use Doctrine\ORM\EntityManager;
+use Laminas\Log\LoggerAwareInterface;
+use Laminas\Session\Container;
+use VuFind\Db\Entity\PluginManager as EntityPluginManager;
+use VuFind\Db\Entity\Resource;
+use VuFind\Db\Entity\User;
+use VuFind\Db\Entity\UserList;
+use VuFind\Db\Entity\UserResource;
+use VuFind\Exception\ListPermission as ListPermissionException;
+use VuFind\Exception\LoginRequired as LoginRequiredException;
+use VuFind\Exception\MissingField as MissingFieldException;
+use VuFind\Exception\RecordMissing as RecordMissingException;
+use VuFind\Log\LoggerAwareTrait;
+use VuFind\Tags;
+
+use function is_object;
+
+/**
+ * Database service for UserList.
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+class UserListService extends AbstractService implements LoggerAwareInterface, ServiceAwareInterface
+{
+    use LoggerAwareTrait;
+    use ServiceAwareTrait;
+
+    /**
+     * Tag parser.
+     *
+     * @var Tags
+     */
+    protected $tagParser;
+
+    /**
+     * Session container for last list information.
+     *
+     * @var Container
+     */
+    protected $session = null;
+
+    /**
+     * Constructor
+     *
+     * @param EntityManager       $entityManager       Doctrine ORM entity manager
+     * @param EntityPluginManager $entityPluginManager VuFind entity plugin manager
+     * @param Tags                $tagParser           Tag parser
+     * @param Container           $session             Session container
+     */
+    public function __construct(
+        EntityManager $entityManager,
+        EntityPluginManager $entityPluginManager,
+        Tags $tagParser,
+        Container $session = null
+    ) {
+        parent::__construct($entityManager, $entityPluginManager);
+        $this->tagParser = $tagParser;
+        $this->session = $session;
+    }
+
+    /**
+     * Get an array of resource tags associated with the list.
+     *
+     * @param UserList $list UserList object.
+     *
+     * @return array
+     */
+    public function getResourceTags($list)
+    {
+        $user = $list->getUser();
+        $tags = $this->getDbService(\VuFind\Db\Service\TagService::class)
+            ->getListTagsForUser($user, null, $list);
```

Comment:
I think we should rename the getListTagsForUser method -- "list tags" are tags on the lists themselves, but this method returns tags on items INSIDE lists. I think the name will cause confusion. Maybe we should rename it to getUserTagsFromFavorites for clarity.